dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyRolesControllerTest.kt

```diff
@@ -282,4 +288,123 @@ class CompanyRolesControllerTest {
         }
         assertErrorCodeInCommunityManagerClientException(headExceptionForNonExistingCompanyOwners, 404)
     }
+
+    @Test
+    fun `assure that a company owner without admin rights can add and remove every company roles`() {
+        val companyId = uploadCompanyAndReturnCompanyId()
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Admin)
+        assignCompanyRole(CompanyRole.CompanyOwner, companyId, dataReaderUserId)
+
+        enumValues<CompanyRole>().forEach {
+            jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
+            assignCompanyRole(it, companyId, dataUploaderUserId)
+
+            jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+            assertDoesNotThrow { hasUserCompanyRole(it, companyId, dataUploaderUserId) }
+
+            jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
+            removeCompanyRole(it, companyId, dataUploaderUserId)
+
+            val exceptionWhenCheckingIfUserIsCompanyOwner = assertThrows<ClientException> {
+                hasUserCompanyRole(it, companyId, dataUploaderUserId)
+            }
+            assertErrorCodeInCommunityManagerClientException(exceptionWhenCheckingIfUserIsCompanyOwner, 404)
+        }
+    }
+
+    @Test
+    fun `assure that a company member admin without admin rights can only add and remove members and member admins`() {
```

Comment:
Auch hier würde ich vorschlagen "without keycloak admin role" zu schreiben.

Reply:
Done

Reply:
Alos slightly changed the wording to be compliant with detek. Please check if the change is fine